### ISSUE TITLE
Functions to set response data in actions

### ISSUE THREAD

**Issue by mrclay** [Elgg/Elgg#9774]
``` php
system_message('Great!');
elgg_set_response_data($data);
forward();
```

Ajax requests would get the data as JSON, otherwise the redirect. Effectively, this eliminates the need to branch for ajax based on `elgg_is_xhr()` and makes it easier to "upgrade" existing actions to returning structured data for ajax.

Sort of an alternative/compliment to #9773

**Comment by mrclay** [comment]
Alternately we could output buffer actions, and consider any output before a forward() as the data, though I'm not sure of the BC implications.

**Comment by hypeJunction** [comment]
See #9779

**Comment by jdalsem** [comment]
Fixed as part of #9960
